### ISSUE TITLE
Bitmart fetchOHLCV not providing history

### ISSUE THREAD

**Issue by xmatthias** [ccxt/ccxt#19888]
### Operating System

linux

### Programming Languages

python

### CCXT Version

4.1.40

### Description

When using the fetch_ohlcv command to get history (using since) - ccxt is still providing only the most recent history.

### Code

``` python
import ccxt
from datetime import datetime
exchange = ccxt.bitmart()
_ = exchange.load_markets()

exchange.verbose = True
since_ms = int(datetime(2023, 1, 1).timestamp() * 1000)
print(since_ms)
x = exchange.fetch_ohlcv('BTC/USDT', timeframe='1m',
                         limit=200, since=since_ms,
                         )

print(f"now: {datetime.now()}")
print(f"last date: {datetime.fromtimestamp(x[-1][0] // 1000)}")
```

after a quick look at the [api docs](https://developer-pro.bitmart.com/en/spot/#get-history-k-line-v3) - it seems like ccxt doesn't make the distinction of endpoints if we're going to "current" [get-latest-k-line-v3](https://developer-pro.bitmart.com/en/spot/#get-latest-k-line-v3) or "historic" data  ([get-history k-line v3](https://developer-pro.bitmart.com/en/spot/#get-history-k-line-v3)).
That's not per se a problem - though the "current" endpoint would return 1000 candles based on the docs (it' doesn't), while the history only returns 200.

While the endpoint used is the "historic" one - the time-based parameters ccxt is sending are wrong - which should be "before" and "after" - but ccxt sends them as "from, to" - which seems to correspond to the v1 api.

### Output:

```

fetch Request: bitmart GET https://api-cloud.bitmart.com/spot/quotation/v3/klines?symbol=BTC_USDT&step=1&from=1672527599&to=1672539599 RequestHeaders: {'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: None

fetch Response: bitmart GET https://api-cloud.bitmart.com/spot/quotation/v3/klines?symbol=BTC_USDT&step=1&from=1672527599&to=1672539599 200 ResponseHeaders: {'Date': 'Thu, 09 Nov 2023 05:12:46 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Vary': 'Origin, Access-Control-Request-Method, Access-Control-Request-Headers', 'X-BM-RateLimit-Limit': '10', 'X-BM-RateLimit-Remaining': '1', 'X-BM-RateLimit-Reset': '2', 'X-Frame-Options': 'SAMEORIGIN', 'Via': '1.1 google', 'CF-Cache-Status': 'DYNAMIC', 'Set-Cookie': '__cf_bm=7iFGYxqH3uO865SJ6KsX32FqPQgpA0LYTVLQRi0W_vM-1699506766-0-ASZJfc4pMkevoefeYh30FFo8Z/svxCP+TNG19SmK1k+4NaGiKezrxp8ihLGK9UEA8ZWhO3iD42k6fjCs5HeokX8=; path=/; expires=Thu, 09-Nov-23 05:42:46 GMT; domain=.bitmart.com; HttpOnly; Secure; SameSite=None, _cfuvid=NTqGvX_8jshc_JZZ2B1As6128PnKnfZGy1q_vGSq3hI-1699506766606-0-604800000; path=/; domain=.bitmart.com; HttpOnly; Secure; SameSite=None', 'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=Oc0X2qnBOUpObZ5oyY%2FI9KD4JfSGvBYSXIcEN5L5KpFsZWRaJlgj7f0WfLN2sqMRNMUkG4k3E70dUz%2FoHARwXPwTJCAgRfPw%2BH5rcDeTw9QachCdbzhqTRTiMJfAY8lPdtvJkdlBcw%3D%3D"}],"group":"cf-nel","max_age":604800}', 'NEL': '{"success_fraction":0.01,"report_to":"cf-nel","max_age":604800}', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains; preload', 'Server': 'cloudflare', 'CF-RAY': '823390096a1a6d92-MUC', 'Content-Encoding': 'gzip'} ResponseBody: {"code":1000,"message":"success","data":[["1699500780","36487.80","36490.00","36475.18","36490.00","4.41564","161098.36"],["1699500840","36490.99","36491.03","36469.18","36491.01","7.34232","267880.73"],["1699500900","36491.14","36521.99","36491.09","36491.09","26.37798","962820.94"],["1699500960","36491.00","36491.00","36470.51","36475.17","7.93912","289603.99"],["1699501020","36475.17","36475.17","36433.13","36436.92","6.57009","239492.03"],["1699501080","36437.14","36443.21","36420.01","36427.07","2.36746","86241.33"],["1699501140","36427.07","36431.46","36409.41","36422.11","6.23956","227242.04"],["1699501200","36418.13","36457.04","36418.13","36457.04","7.00162","255144.92"],["1699501260","36457.03","36483.70","36445.01","36483.70","6.18074","225339.61"],["1699501320","36483.61","36497.68","36473.13","36494.72","7.12105","259851.04"],["1699501380","36493.93","36514.98","36483.57","36485.89","10.68364","389913.70"],["1699501440","36484.93","36491.17","36459.36","36486.32","6.43521","234748.81"],["1699501500","36486.33","36498.03","36471.81","36475.31","8.88002","323974.11"],["1699501560","36477.53","36500.68","36477.53","36493.82","3.34497","122060.34"],["1699501620","36493.82","36493.82","36465.63","36465.64","6.32316","230638.05"],["1699501680","36463.00","36479.99","36458.95","36470.53","2.95621","107822.14"],["1699501740","36470.55","36470.55","36353.61","36375.02","103.04189","3748919.18"],["1699501800","36375.02","36389.93","36371.10","36378.98","31.59624","1149491.73"],["1699501860","36386.48","36399.20","36375.72","36393.20","12.88801","468941.05"],["1699501920","36393.20","36393.21","36375.72","36380.73","8.70686","316811.75"],["1699501980","36376.42","36389.68","36357.39","36378.43","5.32291","193618.73"],["1699502040","36378.44","36393.11","36373.62","36385.60","3.41551","124261.66"],["1699502100","36385.61","36389.67","36377.69","36381.65","5.11742","186174.43"],["1699502160","36382.03","36395.00","36382.03","36385.01","2.59736","94514.99"],["1699502220","36385.01","36398.30","36378.44","36391.23","5.35064","194710.72"],["1699502280","36394.42","36425.35","36394.42","36424.48","10.01144","364501.54"],["1699502340","36424.49","36442.78","36424.48","36442.32","5.48988","200007.86"],["1699502400","36440.43","36494.17","36438.38","36457.80","22.14235","807686.59"],["1699502460","36457.81","36505.00","36457.81","36501.00","16.36465","596945.10"],["1699502520","36497.69","36505.00","36471.10","36480.44","8.70103","317505.36"],["1699502580","36484.13","36490.01","36474.95","36479.70","5.30852","193669.22"],["1699502640","36479.71","36487.90","36473.63","36487.90","10.38496","378871.97"],["1699502700","36487.87","36500.00","36482.65","36482.65","4.28997","156545.98"],["1699502760","36482.65","36482.65","36451.57","36463.45","4.11122","149929.00"],["1699502820","36463.45","36476.92","36454.65","36476.92","5.19117","189295.43"],["1699502880","36475.77","36499.10","36475.77","36499.10","2.97196","108438.83"],["1699502940","36499.09","36513.56","36499.09","36511.40","7.90029","288412.85"],["1699503000","36511.41","36512.89","36504.61","36510.00","8.26573","301778.26"],["1699503060","36513.18","36540.90","36502.20","36502.21","8.53586","311716.14"],["1699503120","36501.44","36507.33","36490.01","36495.12","11.66728","425845.30"],["1699503180","36495.13","36523.56","36495.13","36514.18","5.27005","192434.94"],["1699503240","36514.17","36544.52","36514.17","36532.52","6.81179","248860.22"],["1699503300","36532.53","36548.00","36510.00","36531.64","13.45616","491561.68"],["1699503360","36531.64","36538.99","36509.03","36509.04","5.63549","205833.43"],["1699503420","36509.03","36509.04","36487.71","36487.71","10.59623","386694.25"],["1699503480","36487.72","36495.20","36481.28","36488.10","7.10107","259109.93"],["1699503540","36486.82","36495.13","36485.24","36495.13","4.91145","179228.61"],["1699503600","36499.91","36509.38","36491.04","36506.85","3.73267","136232.28"],["1699503660","36508.46","36512.37","36495.47","36502.13","1.87891","68589.42"],["1699503720","36501.43","36504.85","36492.00","36492.00","1.49980","54746.29"],["1699503780","36492.01","36510.00","36492.00","36507.96","2.96923","108398.37"],["1699503840","36503.95","36503.96","36415.01","36448.50","25.53703","930394.84"],["1699503900","36448.50","36509.02","36441.11","36500.21","98.68553","3601180.46"],["1699503960","36494.68","36499.99","36483.80","36483.81","2.82848","103215.22"],["1699504020","36483.80","36551.01","36474.43","36525.05","19.12241","698380.75"],["1699504080","36525.05","36587.39","36525.05","36579.16","22.23409","813021.19"],["1699504140","36579.17","36617.74","36564.30","36567.07","18.39640","673040.23"],["1699504200","36571.68","36597.08","36559.21","36559.22","7.52595","275266.97"],["1699504260","36556.40","36561.57","36543.48","36558.92","6.78265","247938.19"],["1699504320","36558.92","36588.00","36558.92","36586.94","7.46250","272949.82"],["1699504380","36587.03","36599.00","36581.11","36599.00","4.45001","162827.89"],["1699504440","36600.98","36646.60","36592.38","36635.57","13.26828","485834.88"],["1699504500","36635.55","36671.50","36625.80","36650.21","16.17602","592843.61"],["1699504560","36652.31","36763.33","36650.00","36669.42","116.54356","4276979.86"],["1699504620","36661.20","36746.00","36656.15","36723.15","26.38072","968070.22"],["1699504680","36723.16","36729.69","36691.98","36708.72","14.32403","525854.86"],["1699504740","36708.71","36769.99","36701.72","36767.57","12.51839","459997.34"],["1699504800","36768.17","36831.95","36768.17","36820.32","60.81744","2238527.97"],["1699504860","36812.80","36850.00","36754.59","36756.00","40.63928","1495133.59"],["1699504920","36758.70","36795.47","36757.40","36757.40","21.64888","795984.99"],["1699504980","36757.40","36766.54","36719.73","36752.68","14.08818","517574.28"],["1699505040","36752.63","36752.63","36716.57","36725.00","8.11871","298219.06"],["1699505100","36725.00","36735.82","36669.70","36689.97","21.35699","783580.69"],["1699505160","36684.04","36702.74","36670.83","36688.61","7.35439","269826.93"],["1699505220","36688.61","36716.88","36667.23","36687.57","16.20527","594491.82"],["1699505280","36687.58","36748.99","36687.58","36748.99","12.91686","474308.35"],["1699505340","36755.67","36799.00","36749.31","36792.20","22.50908","827938.69"],["1699505400","36795.04","36816.29","36768.43","36775.00","13.53275","497785.00"],["1699505460","36774.99","36774.99","36726.02","36754.36","6.75206","248157.45"],["1699505520","36754.36","36775.00","36729.85","36729.85","11.52975","423801.49"],["1699505580","36729.85","36738.16","36688.62","36710.17","16.07044","589938.84"],["1699505640","36710.17","36729.98","36688.61","36688.61","9.54008","350200.52"],["1699505700","36688.60","36707.78","36687.14","36698.54","7.17191","263160.56"],["1699505760","36697.58","36710.12","36690.25","36690.25","5.01066","183874.21"],["1699505820","36688.76","36704.24","36685.70","36699.26","7.01701","257497.01"],["1699505880","36699.25","36727.59","36699.25","36716.00","6.79198","249395.46"],["1699505940","36715.99","36725.44","36695.49","36705.98","3.89002","142796.20"],["1699506000","36705.05","36750.00","36705.05","36716.02","6.89725","253370.20"],["1699506060","36712.56","36716.22","36655.50","36655.50","18.83252","690707.08"],["1699506120","36652.97","36669.36","36599.99","36599.99","52.88921","1936547.13"],["1699506180","36599.99","36645.22","36599.99","36625.34","55.27916","2024411.52"],["1699506240","36632.45","36641.44","36591.98","36617.35","11.36932","416225.33"],["1699506300","36618.26","36636.53","36580.94","36608.11","17.18459","629043.16"],["1699506360","36615.35","36628.31","36589.19","36600.00","9.95414","364366.86"],["1699506420","36605.49","36633.15","36605.49","36617.71","7.91538","289855.27"],["1699506480","36621.09","36646.50","36621.09","36646.50","3.43836","125961.75"],["1699506540","36641.28","36641.28","36606.17","36609.63","8.19383","300123.29"],["1699506600","36613.27","36617.69","36600.00","36611.60","4.97334","182072.31"],["1699506660","36611.60","36611.60","36592.54","36599.71","10.16140","371915.22"],["1699506720","36599.70","36599.71","36590.01","36591.19","2.48828","91058.45"]],"trace":"6591fc7b508845359d5fa442e3b3a4fb.74.16995067664675609"}

NOW: 2023-11-09 06:20:01.034559
last date: 2023-11-09 06:19:00
100

```

**Comment by Dan-krm** [comment]
Hi @xmatthias thanks for reporting this, I opened a pull request that will use the new v3 "current" endpoint for spot markets with fetchOHLCV. The max limit for both of those new endpoints is 200 but it looks like the "current" endpoint is a bit faster and has a rate limit that allows more requests per second so I opted for that one. The since argument and also the until parameter should work once the pull request is merged

**Comment by xmatthias** [comment]
Hi @Dan-krm,

Without looking at the pr right now - are you doing the endpoint selection based on the value (or presence) of the since argument?

I don't think the current endpoint provides history past 1000 candles in the past.

**Comment by Dan-krm** [comment]
> I don't think the current endpoint provides history past 1000 candles in the past.

Right we should use the "historic" endpoint then, I opened a separate PR with the change, let me know if there is anything else
